hidimstat/noise_std.py

```diff
@@ -1,66 +1,259 @@
+               eps=1e-2, tol=1e-4, max_iter=1e+4, n_jobs=1, seed=0):
+
+    """Estimation of the covariance matrix using group Reid procedure
+
+    Parameters
+    -----------
+    X : ndarray or scipy.sparse matrix, shape (n_samples, n_features)
+        Data.
+
+    Y : ndarray, shape (n_samples, n_targets)
+        Target.
+
+    fit_Y : bool, optional (default=True)
+        If True, Y will be regressed against X by MultiTaskLassoCV
+        and the covariance matrix is estimated on the residuals.
+        Otherwise, covariance matrix is estimated directly on Y.
+
+    stationary : bool, optional (default=True)
+        If True, noise is considered to have the same magnitude at each
+        time point. Otherwise, magnitude of the noise is not constant.
+
+    method : bool, optional (default='simple')
+        If 'simple', the correlation matrix is estimated by taking the
+        median of the correlation between two consecutive time points
+        and the noise standard deviation at each time point is estimated
+        by taking the median of the standard deviations for every time steps.
+        If 'AR', the order of the AR model is given by `order` and the
+        Yule-Walker method is used to estimate the covariance matrix.
+
+    eps : float, optional (default=1e-2)
+        Length of the cross-validation path.
+        eps=1e-2 means that alpha_min / alpha_max = 1e-2.
+
+    tol : float, optional (default=1e-4)
+        The tolerance for the optimization: if the updates are
+        smaller than ``tol``, the optimization code checks the
+        dual gap for optimality and continues until it is smaller
+        than ``tol``.
+
+    max_iter : int, optional (default=1e+4)
+        The maximum number of iterations.
+
+    n_jobs : int or None, optional (default=1)
+        Number of CPUs to use during the cross validation.
+
+    seed: int, optional (default=0)
+        Seed passed in the KFold object which is used to cross-validate
+        LassoCV. This seed controls the partitioning randomness.
+
+    Returns
+    -------
+    sigma_hat : float
+        Estimated noise standard deviation.
+
+    beta_hat : array, shape (n_features,)
+        Estimated parameter vector.
+    """
+
+    X = np.asarray(X)
+    n_samples, n_features = X.shape
+    n_targets = Y.shape[1]
+
+    if method == 'simple':
+        print('Group reid: simple cov estimation')
+    else:
+        print('Group reid: ' + method + str(order) + ' cov estimation')
+
+    if (max_iter // 5) <= n_features:
+        max_iter = n_features * 5
+        print("'max_iter' has been increased to {}".format(max_iter))
+
+    cv = KFold(n_splits=5, shuffle=True, random_state=seed)
+
+    if fit_Y:
+
+        clf_mtlcv = \
+            MultiTaskLassoCV(eps=eps, normalize=False, fit_intercept=False,
+                             cv=cv, tol=tol, max_iter=max_iter,
+                             n_jobs=n_jobs)
+
+        clf_mtlcv.fit(X, Y)
+        Beta_hat = clf_mtlcv.coef_
```

Comment:
please not capital letters in var names. The only exception is for X and Y
```suggestion
        beta_hat = clf_mtlcv.coef_
```